File: 0495-teemo-attacking/0495-teemo-attacking.java
class Solution {
    public int findPoisonedDuration(int[] timeSeries, int duration) {
        if (timeSeries == null || timeSeries.length == 0) return 0;
        int tem=0;
        for (int i=0;i<timeSeries.length-1;i++) {
            tem += Math.min(timeSeries[i+1]-timeSeries[i],duration);
        }
        return tem+duration;          
    }
}

File: 0495-teemo-attacking/0495-teemo-attacking.c
#include <stddef.h>

int findPoisonedDuration(int *timeSeries, int n, int duration) {
    int ans = 0;
    int i;
    if (n == 0) {
        return 0;
    }
    for (i = 1; i < n; i++) {
        int gap = timeSeries[i] - timeSeries[i - 1];
        ans += (duration < gap) ? duration : gap;
    }
    ans += duration;
    return ans;
}